src/components/header/navbar/navbar.tsx

```diff
@@ -81,7 +81,7 @@ function showLinks(userName: string | undefined) {
 }
 
 const NavBar = () => {
-  const userName = useSelector((state) => (state as User).userName);
+  const userName = useSelector((state) => (state as RootState).user.userName);
```

Comment:
try ` const userName = useSelector((state: RootState) => state.user.userName);`

Reply:
rewritten as you'd adviced, thank you